eos-image-boot-dm-setup

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# When booting from an image file hosted on a filesystem, this program maps the
+# host filesystem to a dm-device with a hole where the image file lives.
+
+for i in $(</proc/cmdline); do
+  case $i in
+  endless.image.device=*)
+    host_device=${i//endless.image.device=}
```

Comment:
The generator runs in the initrd, and the initrd's `/tmp` does not persist through the switchroot. I haven't gone as far as checking why this is the case, but the file just isn't there.